### PR TITLE
Fix "-" and unset handling in zeek writer

### PR DESCRIPTION
**Pull request by henridf**

### CODE REVIEW

File: zbuf/zval.go [hunk 1]
```diff
@@ -73,28 +72,33 @@ func ZvalToZeekString(typ zng.Type, zv zcode.Bytes, isContainer bool, utf8 bool)
...
+
 			if it.Done() {
 				break
 			}
 			_ = b.WriteByte(',')
 		}
-		s = b.String()
+		return b.String()
+	case *zng.TypeOfString:
+		if len(zv) == 1 && zv[0] == '-' {
```
Comment (nwt) [inline zbuf/zval.go]: Nit: Maybe `string(zq) == "-"` if you find it clearer.

Comment (henridf) [reply zbuf/zval.go]: I do, but thought that might allocate.

**Comment by henridf** [comment]
> I think ZvalToZeekString deserves a unit test.

I agree! I'll add one before merging.

File: zbuf/zval.go [hunk 2]
```diff
@@ -73,28 +72,33 @@ func ZvalToZeekString(typ zng.Type, zv zcode.Bytes, isContainer bool, utf8 bool)
 			if err != nil {
 				return "error in ZvalToZeekString"
 			}
-			val, err := inner.New(v)
+			fld := ZvalToZeekString(inner, v, utf8)
 			if err != nil {
 				return "error in ZvalToZeekString"
 			}
```
[the remark this answers is not in the document]
Comment (henridf) [reply zbuf/zval.go]: d'oh

File: zbuf/zval.go (hunk 1 above)
Comment (nwt) [reply zbuf/zval.go]: It does not.

**Comment by henridf** [comment]
> I think ZvalToZeekString deserves a unit test.

Added

**Pull request merged**